server/src/main/java/org/candlepin/resource/EntitlementResource.java

```diff
@@ -152,18 +125,24 @@ public EntitlementDTO hasEntitlement(@PathParam("consumer_uuid") String consumer
                 consumerUuid, productId));
     }
 
-    @ApiOperation(notes = "Retrieves list of Entitlements", value = "listAllForConsumer")
-    @ApiResponses({ @ApiResponse(code = 400, message = "") })
-    @GET
-    @Produces(MediaType.APPLICATION_JSON)
+    @Override
     public List<EntitlementDTO> listAllForConsumer(
-        @QueryParam("consumer") String consumerUuid,
-        @QueryParam("matches") String matches,
-        @QueryParam("attribute") List<KeyValueParameter> attrFilters,
-        @Context PageRequest pageRequest) {
+        String consumerUuid,
+        String matches,
+        List<KeyValueParamDTO> attrFilters,
+        Integer page,
+        Integer perPage,
+        String order,
+        String orderBy) {
 
         EntitlementFilterBuilder filters = EntitlementFinderUtil.createFilter(matches, attrFilters);
         Page<List<Entitlement>> p;
+        PageRequest pageRequest = PageRequest.create(() -> this.i18n)
```

Comment:
One thing to keep in mind with how paging works today, is that the presence of paging info often results in at least one additional query (to get counts for the headers). So we may not want to default to the always-create behavior we have here.